### ISSUE TITLE
Contacts Typing/Replying Notifier or Indicator

### ISSUE THREAD

**Issue by AmasterAmaster** [Yellow-Dog-Man/Resonite-Issues#3209]
### Is your feature request related to a problem? Please describe.

This will be blocked by #1116. However, I want this to be tracked as I feel this would help ease the awkwardness of replying to contacts on both ends (such as waiting for someone to respond, late replying, or seeing that they read/heard the message and are replying back).

Currently in Resonite, the awkwardness comes from needing to reply or waiting for a reply from someone after sending a message. And knowing that you are not left on read and seeing that someone is actually trying to respond is nice to have. However, there are many situations and special cases that I am aware of where it becomes very difficult to respond, such as:
- Loading into another world and being forced to load assets and not being able to respond in a timely manner (or in a way to let users know that your loading in a world, maybe by hiding the typing/replying indicator since typing stopped?)
- Laggy worlds, or worlds that have low FPS, this makes it difficult to let users know that your replying to them since typing on vr is super slow in cases like this, but having a typing/replying indicator would help users know you are trying to reply back to them.
- World crashes, or in an event where Resonite is unresponsive, having a typing/replying indicator may help to show the user that you were typing earlier, even after said indicator disappears, it at least shows that you attempted to reply.

Besides those special cases, replying to someone should not be a waiting game or a "[game of chicken](https://en.wikipedia.org/wiki/Chicken_(game))".
I would also imagine that the act of typing or the act of replying with a voice message would trigger it, but that is up to you all on how to implement that.

### Describe the solution you'd like

A typing/replying indicator to let me know when someone is replying to me, and vice versa. Similar to how discord handles typing/replying to someone. The implementation is up to you.

### Describe alternatives you've considered

Considered these things: 
- Just using discord instead of using Resonite for communication.
- Just dealing with the awkwardness of not knowing if I was on read or just waiting for a reply when the other user is never going to reply (the awkward wait...)

### Additional Context

This does not account for social awkwardness, or any social normalities that come from messaging other users. This only accounts for the functionality of letting other users know that you are trying to reply to them.

### Requesters

- AmasterAmaster

**Comment by Frooxius** [comment]
This is something I actually did want to add at some point as part of the contacts UI overhaul.

**Comment by Dusty-Sprinkles** [comment]
If this does get implemented, an opt-out would be nice as well for those who don't really want that (personally I've had more games of chicken with "x is typing", stopping typing till they respond, and vise versa than just waiting for a response from someone)

**Comment by Frooxius** [comment]
Yes, I was thinking same, have this with an opt out.
